tests/conftest.py

```diff
@@ -4,73 +4,148 @@
+            table_name='thread',
+            schema_name='swoop',
+            columns=[
+                'created_at',
+                'last_update',
+                'action_uuid',
+                'handler_name',
+                'priority',
+                'status',
+                'next_attempt_after',
+                'error'
+            ],
+            records=[
+                (created,created,'81842304-0aa9-4609-89f0-1c86819b0752',
+                 'handler',100,'PENDING',retry_at,'none'),
+                (completed,completed,'2595f2da-81a6-423c-84db-935e6791046e',
+                 'handler',100,'SUCCESSFUL',retry_at,'none')
+            ],
+        )
+
+        events = await conn.copy_records_to_table(
+            table_name='event',
+            schema_name='swoop',
+            columns=[
+                'event_time',
+                'action_uuid',
+                'status',
+                'event_source',
+                'retry_seconds',
+                'error'
+            ],
+            records=[
+            #(created,'2595f2da-81a6-423c-84db-935e6791046e','PENDING','swoop-db',300,'none'),
+            (queued,'2595f2da-81a6-423c-84db-935e6791046e','QUEUED','swoop-db',300,'none'),
+            (started,'2595f2da-81a6-423c-84db-935e6791046e','RUNNING','swoop-db',300,'none'),
+            (completed,'2595f2da-81a6-423c-84db-935e6791046e','SUCCESSFUL','swoop-db',300,'none')
+            ],
+        )
+
+        print (f'Inserted Actions: {actions}')
+        print (f'Inserted Threads: {threads}')
+        print (f'Inserted Events: {events}')
+
+
+async def drop_database(db_name: str, db_connection_string: str) -> None:
+    async with get_db_connection(db_connection_string) as conn:
+        await conn.execute(f'DROP DATABASE "{db_name}";')
 
 
-async def async_drop_database(db_name: str, db_connection_string: str) -> None:
-    conn = None
-    try:
-        conn = await asyncpg.connect(db_connection_string)
-        await conn.execute(f'DROP DATABASE "{db_name}";')
-    finally:
-        if conn:
-            await conn.close()
+# If using pytest_asyncio fixtures, the event_loop scope must at least match
+# the scope of those fixtures (currently "module")
+@pytest.fixture(scope="module")
+def event_loop():
+    loop = asyncio.get_event_loop()
+    yield loop
+    loop.close()
 
 
-def drop_database(db_name: str, db_connection_string: str) -> None:
-    asyncio.run(async_drop_database(db_name, db_connection_string))
+@pytest.fixture(scope="session")
+def dbschema(pytestconfig) -> Path:
+    return pytestconfig.rootpath.joinpath("db", "schema.sql")
 
 
+# Will set a unique database name (settings.database_url) per module
 @pytest.fixture(scope="module")
-def database(dbschema, settings):
+def settings() -> Settings:
+    return Settings()
+
+
+@pytest_asyncio.fixture(scope="module")
+async def test_app(database):
+    app.state.settings.database_url = database
+    return app
```

Comment:
I'm not loving the idea of this being module scope. Now tests cannot modify the app state in an isolated way.